security/validateMessage.go

```diff
@@ -2,79 +2,146 @@ package security
-		log.Printf("timestamp unvalid : got %d but need to be between %d and %d",
-			data.Creation, time.Now().Unix()-MaxClockDifference, time.Now().Unix()+MaxPropagationDelay)
+	creation := checkCreationTimeValidity(data.Creation)
+
+	//we receive either start either scheduled and the other is set to 0
+	if data.Start != 0 && data.Scheduled != 0 {
+		log.Printf("cannot have both start and scheduled set")
+		return false
+	} else if data.Start == 0 && data.Scheduled == 0 {
+		log.Printf("at least one of Data and Schedule has to be set")
 		return false
 	}
 
-	//we start after the creation and we end after the start
-	if data.Start < data.Creation || data.End < data.Start {
-		log.Printf("timestamps not logic. Either end is before start, or start before creation.")
-		return false
+	if data.Scheduled == 0 {
+		//we start after the creation and we end after the start
+		if data.Start < data.Creation {
+			log.Printf("timestamps not logic. Start cannot be before creation.")
+			return false
+		}
+	} else if data.Start == 0 {
+		//we start after the creation and we end after the start
+		if data.Scheduled < data.Creation {
+			log.Printf("timestamps not logic. Scheduled cannot be before creation.")
+			return false
+		}
 	}
+
 	//need to meet some	where
-	if data.Location == "" {
-		log.Printf("location can not be empty")
+	location := checkStringNotEmpty(data.Location)
+	//name cannot be empty
+	name := checkStringNotEmpty(data.Name)
+
+	return name && creation && location && checkRollCallId(laoId, data.Creation, data.Name, data.ID)
+}
+
+//checkRollCallId check if id is correct  : SHA256('R'||lao_id||creation||name)
+func checkRollCallId(laoId string, creation int64, name string, id []byte) bool {
+	var elementsToHashForDataId []string
+	elementsToHashForDataId = append(elementsToHashForDataId, "R", b64.StdEncoding.EncodeToString([]byte(laoId)), strconv.FormatInt(creation, 10), name)
+	hash := HashOfItems(elementsToHashForDataId)
+	if !bytes.Equal(id, hash) {
+		log.Printf("ID of RollCall invalid: %v should be: %v", string(id), string(hash[:]))
 		return false
 	}
 	return true
 }
 
-func PollCreatedIsValid(data message.DataCreatePoll, message message.Message) bool {
-	return true
+//RollCallOpenedIsValid tell if a Roll call is valid on opening or reopening
+func RollCallOpenedIsValid(data message.DataOpenRollCall, laoId string, rollCall event.RollCall) bool {
+	//we start after the creation and we end after the start
+	if data.Start < rollCall.Creation {
+		log.Printf("timestamps not logic.Start before creation.")
+		return false
+	}
+	return checkRollCallId(laoId, rollCall.Creation, rollCall.Name, data.ID)
 }
 
-func RollCallCreatedIsValid(data message.DataCreateRollCall, message message.Message) bool {
-	return true
+//RollCallClosedIsValid tell if a rollCall timestamps make sense
+func RollCallClosedIsValid(data message.DataCloseRollCall, laoId string, rollCall event.RollCall) bool {
```

Comment:
concerning the roll Call in general, more verification shoud be done but we sticked to the protospec